app/javascript/components/workflows/professor/exams/accommodations/index.tsx

```diff
@@ -0,0 +1,326 @@
+      refresh();
+      alert({
+        variant: 'success',
+        title: 'Successfully updated accommodation',
+        message: `Accommodation for '${accommodation.reg.user.displayName}' updated.`,
+      });
+    }).catch((err) => {
+      setEditing(false);
+      alert({
+        variant: 'danger',
+        title: 'Error updating accommodation',
+        message: err.message,
+      });
+    });
+  };
+  if (editing) {
+    return (
+      <AccommodationEditor
+        accommodation={accommodation}
+        cancel={stopEdit}
+        submit={submit}
+      />
+    );
+  }
+  return (
+    <tr>
+      <td className="align-middle">
+        {accommodation.reg.user.displayName}
+      </td>
+      <td className="align-middle">
+        {accommodation.startTime ? (
+          <ReadableDate showTime value={accommodation.startTime} />
+        ) : (
+          <i>Not set.</i>
+        )}
+      </td>
+      <td className="align-middle">{accommodation.extraTime}</td>
+      <td align="right" className="text-nowrap">
+        <Button
+          variant="danger"
+          onClick={destroy}
+        >
+          <Icon I={FaTrash} />
+          <span className="ml-2">
+            Delete
+          </span>
+        </Button>
+        <Button
+          variant="primary"
+          className="ml-2"
+          onClick={edit}
+        >
+          <Icon I={BsPencilSquare} />
+          <span className="ml-2">
+            Edit
+          </span>
+        </Button>
+      </td>
+    </tr>
+  );
+};
+
+interface Selection {
+  label: string;
+  value: number;
+}
+
+const NewAccommodation: React.FC<{
+  accommodations: Accommodation[];
+  refresh: () => void;
+}> = (props) => {
+  const {
+    accommodations,
+    refresh,
+  } = props;
+  const { examId } = useParams();
+  const res = useRegistrationsIndex(examId);
+  const [selected, setSelected] = useState<Selection>(null);
+  const { alert } = useContext(AlertContext);
+  const accs = new Set();
+  accommodations.forEach((a) => accs.add(a.reg.id));
+  const options = res.type === 'RESULT'
+    ? res.response.registrations
+      .sort((a, b) => a.displayName.localeCompare(b.displayName))
+      .map((r) => ({
+        label: r.displayName,
+        value: r.id,
+      }))
+      .filter((r) => !accs.has(r.value))
+    : [];
+  const submit = () => {
+    if (!selected) return;
+    createAccommodation(examId, selected.value).then((result) => {
+      if (result.success !== true) {
+        throw new Error(result.reason);
+      }
+      refresh();
+      setSelected(null);
+      alert({
+        variant: 'success',
+        title: 'Successfully created accommodation',
+        autohide: true,
+        message: `Accommodation for '${selected.label}' created.`,
+      });
+    }).catch((err) => {
+      refresh();
+      alert({
+        variant: 'danger',
+        title: 'Error creating accommodation',
+        message: err.message,
+      });
+    });
+  };
+  if (res.type === 'ERROR') {
```

Comment:
You have a bunch of `(res.type === something)` comparisons in this code, rather than the `switch (res.type)` you typically have.

Reply:
This is so we can use the same code to render when loading _or_ finished

Reply:
changed to switch